Sources/TuistCache/Cache/FileDownloader.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+import RxSwift
+import TSCBasic
+
+enum FileDownloaderError: LocalizedError {
+    case noLocalURL
+    case invalidResponse
+    case urlSessionError(Error)
+    case moveFileError(Error)
+
+    var errorDescription: String? {
+        switch self {
+        case .noLocalURL:
+            return "Didn't receive a valid local URL"
```

Comment:
If I was a developer and got this error I wouldn't know what to do. I'd include more details:
```
The download of framework X from the cache succeeded, but the file doesn't exist locally. 
```

Reply:
Good point, addressed.